qemu/tests/qemu_disk_img_commit.py

```diff
@@ -44,7 +44,11 @@ def run(test, params, env):
     commit_test.start_vm(n_params)
 
     # save file md5sum before commit
-    md5 = commit_test.save_file(t_file)
+    guest_sync = False
+    drive_cache = commit_test.params.get("drive_cache")
+    if drive_cache == 'none' or drive_cache == 'writeback':
```

Comment:
After copy the file from host to guest, qemu may be killed by qmp command. The file will not be flushed to disk whatever the drive cache mode. So I think it is better to sync directly.